### ISSUE TITLE
speed control

### ISSUE THREAD

**Issue by ioanvapi** [jdarling/pxt-pca9685#8]
Hi. Is there a way to control the rotation speed of an ordinary servo? I see you have functions to control the speed for continuous rotation servos.

**Comment by jdarling** [comment]
Unfortunately that would be impossible without some type of feedback line from the servo that reported the current position (exactly) and some timers to set a new position every (n) interval.

You could come close by just keeping the current position in memory (the library already does this for you once it has been set) and then just changing the desired location and waiting for an interval before setting the next location.

EG: If you are at 0 and want to go to 90 in 5 seconds, you would divide 90 by 5 (18) then every second would move 18 degrees toward 90.  So step 1 to 18 deg, wait one second, move to 36 deg, etc...

An ordinary servo moves to it's position based on a PID feedback loop.  To get speed control out of a continuous rotation we exploit this by setting the desired setpoint further away from the current zero point.  Since PID tries to move proportional to the difference this causes the servo to speed up when the distance is further and slow down when the distance is closer.

Hope that explains better.

**Comment by ioanvapi** [comment]
Hi, thanks for your explanation. I guess I got your point. I have figured samething similar but the steps values follwing a bell curve. I mean, at start and at the end the servo will do smaller steps and bigger steps in the middle when the servo will rotate faster. I have just to figure some custom delays between these steps.
I'm thinking to 'extend=copy to a new function' your  setServoPosition() function by adding a new param, a time interval, that will be split between these steps. The one who invokes this function should provide this interval too. Also, I can figure a default value.
Does it sound as a good idea? Thanks.
